src/simpleNbox.cpp

```diff
@@ -998,4 +1132,201 @@ void SimpleNbox::set_c0(double newc0)
         
 }
 
+// Add a biome to a time-series map variable (e.g. veg_c_tv)
+template <class T_data>
+void SimpleNbox::add_biome_to_ts(tvector<std::map<std::string, T_data>>& ts,
+                                 const std::string& biome,
+                                 T_data init_value) {
+    // First, check if a biome of this name already exists in the data
+    if ( ts.get(ts.firstdate()).count( biome ) ) {
+        H_THROW( "Biome '" + biome + "' already exists in data." );
+    }
+    
+    // Loop over time steps, and set the variable to the provided `init_value`
+    std::map<std::string, T_data> newval;
+    for ( double i = ts.firstdate(); i < ts.lastdate(); i++ ) {
+        if (ts.exists(i)) {
+            newval = ts.get(i);
+            newval[ biome ] = init_value;
+            ts.set(i, newval);
+        }
+    }
+}
+
+// Remove a biome from a time-series map variable
+template <class T_map>
+void SimpleNbox::remove_biome_from_ts(tvector<T_map>& ts,
+                                      const std::string& biome) {
+    // We don't need to check for presence of `biome` here because the
+    // `<std::map>.erase()` method is effectively a no-op when given a
+    // non-existent key.
+    T_map currval;
+    for ( double i = ts.firstdate(); i < ts.lastdate(); i++ ) {
+        if (ts.exists(i)) {
+            currval = ts.get(i);
+            currval.erase(biome);
+            ts.set(i, currval);
+        }
+    }
+}
+
+// Remove a biome from a time-series map variable
+template <class T_map>
+void SimpleNbox::rename_biome_in_ts(tvector<T_map>& ts,
+                                    const std::string& oldname,
+                                    const std::string& newname) {
+    if ( !ts.get(ts.firstdate()).count( oldname ) ) {
+        H_THROW( "Biome '" + oldname + "' not found in data.");
+    }
+    if ( ts.get(ts.firstdate()).count( newname ) ) {
+        H_THROW( "Biome '" + newname + "' already exists in data.");
+    }
+
+    T_map currval;
+    for ( double i = ts.firstdate(); i < ts.lastdate(); i++ ) {
+        if (ts.exists(i)) {
+            currval = ts.get(i);
+            currval[newname] = currval.at(oldname);
+            currval.erase(oldname);
+            ts.set(i, currval);
+        }
+    }
+}
+
+// Check if `biome` is present in biome_list
+bool SimpleNbox::has_biome(const std::string& biome) {
+    return std::find(biome_list.begin(), biome_list.end(), biome) != biome_list.end();
```

Comment:
How often do we search `biome_list` for a value (vs. just iterating over whatever happens to be there)?  If it's something we do often, then we might want to use a `std::set<string>` instead of a vector.  If not, then what we have is fine.

Reply:
Is this a basically free performance boost? I.e. The only consequence of changing to `std::set` is that the operation becomes faster (and we get the assurance that entries are unique)? Or are there other side effects I should keep eye out for?

Reply:
So, as I understand it (which may definitely be wrong), one issue with using a `set` here is that we lose the history of biome creation. The main place where this matters is that the current implementation of `SimpleNbox::createBiome` will initialize a biome with the parameters from the most recently set biome. However, I think this could also introduce some minor surprises on the user side if the position of biomes in the biome list cannot be predicted unless you know the contents of the existing biome list.

I'm inclined to keep this as is right now, but I think it's something we could revisit in the future.

Reply:
It's a performance boost for searching for a specific biome, but it's a performance hit for iterating over whatever biomes happen to be there.  And you are correct that a set will not allow you to save the history of the biome creation.  I'm ok with the current implementation.  I was just calling out the alternative for you to evaluate.